src/fluree/server/system.clj

```diff
@@ -198,39 +200,40 @@
       (log/info "  Closed mode: enabled"))))
 
 (defn start-config
-  ([config]
-   (start-config config nil))
-  ([config profile]
-   (let [json-config (if (string? config)
-                       (json/parse config false)
-                       config)
-         config-with-profile (if profile
-                               (config/apply-profile json-config profile)
-                               json-config)
-         parsed-config (config/parse config-with-profile)]
-     (log-config-summary parsed-config)
-     (conn-system/initialize parsed-config))))
+  [config & {:keys [profile reindex]}]
+  (let [json-config         (if (string? config)
+                              (json/parse config false)
+                              config)
+        config-with-profile (if profile
+                              (config/apply-profile json-config profile)
+                              json-config)
+        parsed-config       (config/parse config-with-profile)
+        system              (conn-system/initialize parsed-config)]
+    (log-config-summary parsed-config)
+    (when reindex
+      (let [connection-key (first (filter #(isa? % :fluree.db/connection) (keys system)))]
+        (when-let [conn (get system connection-key)]
```

Comment:
this should be an `if-let`, with the else branch exiting with an error about no connection being found.